### ISSUE TITLE
Don't require style and test packages for every installation

### ISSUE THREAD

**Issue by jblakeman** [localstack/localstack#1323]
I recommend having the option to install things like `flake8`, `flake8-quotes` only when specifying development extras like `localstack[dev]` rather than for every install, which can cause unnecessary package conflicts in the consuming environment.  One way to do this is by splitting dev / test requirements into separate files, consuming them in `setup.py` then defining them in the `extras_require` kwarg.

```python
    install_requires=main_requirements,
    extras_require={
        'dev': dev_requirements
    }
```

**Comment by jblakeman** [comment]
Alternatively, just add the `#extended-lib` annotation to these packages, as noted [here](https://github.com/localstack/localstack/blob/85ae19a565c73feed82304a426ebe5ba044ad802/requirements.txt#L1).

**Comment by englehardt** [comment]
+1 Note that `flake8` automatically imports any installed plugins, and as far as I can tell there's no way to prevent it from doing so: https://gitlab.com/pycqa/flake8/issues/133.

Thus, installing `localstack` into a project that also uses `flake8` for style checking will end up enabling `flake8-quotes` for the entire project.

**Comment by whummer** [comment]
Thanks for reporting @jblakeman @englehardt . This should be fixed in #1903 - the change will become effective when we publish the next release of the pip package (which will happen soon). Thanks
